diff --git a/package/efivar/efivar.mk b/package/efivar/efivar.mk
index 492a4468bf..0a2cd278a2 100644
--- a/package/efivar/efivar.mk
+++ b/package/efivar/efivar.mk
@@ -14,7 +14,8 @@ EFIVAR_INSTALL_STAGING = YES
 # -library.
 EFIVAR_MAKE_OPTS = \
 	libdir=/usr/lib \
-	LDFLAGS="$(TARGET_LDFLAGS) -fPIC"
+	LDFLAGS="$(TARGET_LDFLAGS) -fPIC" \
+	CFLAGS="-Wno-error=stringop-overflow"
 
 define EFIVAR_BUILD_CMDS
 	# makeguids is an internal host tool and must be built separately with
