--- a/arch/arm/boot/dts/bcm2708-rpi-0-w.dts	2018-11-30 23:46:36.656912915 +0800
+++ b/arch/arm/boot/dts/bcm2708-rpi-0-w.dts	2018-11-30 23:50:42.030444798 +0800
@@ -144,8 +144,8 @@
 &leds {
 	act_led: act {
 		label = "led0";
-		linux,default-trigger = "mmc0";
-		gpios = <&gpio 47 0>;
+		linux,default-trigger = "heartbeat";
+		gpios = <&gpio 47 1>;
 	};
 };
 
--- a/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts	2018-11-30 23:46:36.656912915 +0800
+++ b/arch/arm/boot/dts/bcm2708-rpi-b-plus.dts	2018-11-30 23:50:42.030444798 +0800
@@ -90,8 +90,8 @@
 &leds {
 	act_led: act {
 		label = "led0";
-		linux,default-trigger = "mmc0";
-		gpios = <&gpio 47 0>;
+		linux,default-trigger = "heartbeat";
+		gpios = <&gpio 47 1>;
 	};
 
 	pwr_led: pwr {
